src/Build/BuildCheck/Infrastructure/IBuildCheckManager.cs

```diff
@@ -27,6 +27,8 @@ internal enum BuildCheckDataSource
 /// </summary>
 internal interface IBuildCheckManager
 {
+    bool isRestore { get; set; }
```

Comment:
This doesn't seem to be used - it can be safely removed (along with implementations).

Reply:
Oh I see.

In such case - the logger and the worker node can - and should - maintain their own private indication of whether restore is happening. The logger should maintain it per project.

* The worker node is not guaranteed to be part of the entrypoint node (that runs the loggers). Currently only `RequestBuilder` is setting the indication - that would not work for the case (you can experiment with this via the `EndToEndTest` that sets `NOINPROCNODE` in one of the cases to `1` - then the logger runs alone)
* Different projects can run their different phases during the same build - logger cannot determine the restore status globally - only per project. And it will need to get the indication from BuildEventArgs - as there might not be anything else available in the same process.